src/webview/ConnectionPanel.ts

```diff
@@ -0,0 +1,322 @@
+import * as vscode from 'vscode';
+import { ExtensionContext, ViewColumn } from 'vscode';
+import { Connection, ConnectionManager } from '../connectionManager';
+import { ConnectionsViewProvider } from './sideBar';
+
+export enum PanelMode {
```

Comment:
Generally, in TypeScript it's preferred to use a [union type](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) of [string literals](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) over an `enum`:

```javascript
export type PanelMode = "Uninitialized" | "Add" | "Edit"